drand/drand_test.go

```diff
@@ -22,3 +27,159 @@ func TestNew(test *testing.T) {
 		test.Error("dRand should belong to a leader")
 	}
 }
+
+func TestGetValidatorPeers(test *testing.T) {
+	leader := p2p.Peer{IP: "127.0.0.1", Port: "9902"}
+	validator := p2p.Peer{IP: "127.0.0.1", Port: "9905"}
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+
+	if !dRand.IsLeader {
+		test.Error("dRand should belong to a leader")
+	}
+
+	countValidatorPeers := len(dRand.GetValidatorPeers())
+
+	if countValidatorPeers != 2 {
+		test.Error("Count of validator peers doesn't match")
+	}
+}
+
+func TestAddPeers(test *testing.T) {
+	leader := p2p.Peer{IP: "127.0.0.1", Port: "9902"}
+	validator := p2p.Peer{IP: "127.0.0.1", Port: "9905"}
+	priKey, _, _ := utils.GenKeyP2P("127.0.0.1", "9902")
+	host, err := p2pimpl.NewHost(&leader, priKey)
+	if err != nil {
+		test.Fatalf("newhost failure: %v", err)
+	}
+	dRand := New(host, "0", []p2p.Peer{leader, validator}, leader, nil, true)
+
+	if !dRand.IsLeader {
+		test.Error("dRand should belong to a leader")
+	}
+
+	newPeer := p2p.Peer{IP: "127.0.0.1", Port: "9907"}
+	countValidatorPeers := dRand.AddPeers([]*p2p.Peer{&newPeer})
+
+	if countValidatorPeers != 1 {
```

Comment:
can you also verify the total number of peers after the AddPeers call?

Reply:
Done